ai21_tokenizer/tokenizer_factory.py

```diff
@@ -25,23 +21,22 @@ class TokenizerFactory:
     Currently supports only J2-Tokenizer
     """
 
-    _tokenizer_name = PreTrainedTokenizers.J2_TOKENIZER
-
     @classmethod
-    def get_tokenizer(cls) -> BaseTokenizer:
-        config = cls._get_config(cls._tokenizer_name)
-        model_path = cls._model_path(cls._tokenizer_name)
-        return JurassicTokenizer(model_path=model_path, config=config)
+    def get_tokenizer(cls, tokenizer_name: str = PreTrainedTokenizers.J2_TOKENIZER) -> BaseTokenizer:
+        if tokenizer_name == PreTrainedTokenizers.JAMBA_TOKENIZER:
+            return cls._create_jamaba_tokenizer("<huggingface-tokenizer-url-placeholder>")
 
-    @classmethod
-    def _tokenizer_dir(cls, tokenizer_name: str) -> Path:
-        return _LOCAL_RESOURCES_PATH / tokenizer_name
+        if tokenizer_name == PreTrainedTokenizers.J2_TOKENIZER:
+            return cls._create_jurassic_tokenizer(_LOCAL_RESOURCES_PATH / PreTrainedTokenizers.J2_TOKENIZER)
```

Comment:
Worth adding a follow up task for removing this asset from the repo and downloading it same as jamba